### PR TITLE
[PRDP-58] feat: Updated logs for datastore processes

### PR DESCRIPTION
**Pull request by alessio-cialini**
Ref to https://github.com/pagopa/pagopa-infra/pull/1123

#### List of Changes

- Updated logs to enable tracing on eventId in logs

#### Motivation and Context

Required to check for logs related to the appInsight in the receipt datastore process

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public void processBizEventToReceipt(
         List<Receipt> itemsDone = new ArrayList<>();
         Logger logger = context.getLogger();
 
-        String msg = String.format("BizEventEnrichment stat %s function - num events triggered %d", context.getInvocationId(), items.size());
+        String msg = String.format("BizEventToReceipt stat %s function - num events triggered %d", context.getInvocationId(), items.size());
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java:64]: use [context](https://learn.microsoft.com/en-us/java/api/com.microsoft.azure.functions.executioncontext?view=azure-java-stable#method-details) to retrive correct fn name

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public void processBizEventToReceipt(
 
                     receipt.setEventData(eventData);
 
-                    String message = String.format("BizEventToReceipt function called at %s for event with id %s and status %s",
+                    String message = String.format("[BizEventToReceiptProcessor] function called at %s for event with id %s and status %s",
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java:91]: ditto

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ public void processBizEventToReceipt(
                 discarder++;
 
                 //Error info
-                msg = String.format("Error to process event with id %s", bizEvent.getId());
+                msg = String.format("[BizEventToReceiptProcessor] Error to process event with id %s", bizEvent.getId());
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/BizEventToReceipt.java:109]: ditto and add trace on discarder biz event too.
Could be useful for troubleshooting

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 4]
```diff
@@ -91,7 +91,7 @@ public void processGenerateReceipt(
         List<Receipt> itemsToNotify = new ArrayList<>();
         Logger logger = context.getLogger();
 
-        String logMsg = String.format("GenerateReceipt function called at %s", LocalDateTime.now());
+        String logMsg = String.format("[GenerateReceiptProcess] function called at %s for bizEvent with id %s", LocalDateTime.now(), bizEvent.getId());
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:94]: use context fn name

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 5]
```diff
@@ -103,7 +103,7 @@ public void processGenerateReceipt(
         try {
             receipt = receiptCosmosClient.getReceiptDocument(bizEvent.getId());
         } catch (ReceiptNotFoundException e) {
-            String errorMsg = String.format("Receipt not found with the biz-event id %s", bizEvent.getId());
+            String errorMsg = String.format("[GenerateReceiptProcess] Receipt not found with the biz-event id %s", bizEvent.getId());
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:106]: ditto

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 6]
```diff
@@ -119,20 +119,29 @@ public void processGenerateReceipt(
...
-            logger.info(logMsg);
-
             //Verify if debtor's and payer's fiscal code are the same
             String debtorCF = receipt.getEventData().getDebtorFiscalCode();
             String payerCF = receipt.getEventData().getPayerFiscalCode();
 
             if (debtorCF != null || payerCF != null) {
                 boolean generateOnlyDebtor = payerCF == null || payerCF.equals(debtorCF);
 
+                String log = String.format(
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:129]: missing "eventId" biz

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 7]
```diff
@@ -119,20 +119,29 @@ public void processGenerateReceipt(
...
+                        "[GenerateReceiptProcess] Generating pdf for Receipt with id %s",
+                        receipt.getId()
+                );
+                logger.info(log);
+
                 //Generate and save PDF
                 PdfGeneration pdfGeneration = service.handlePdfsGeneration(generateOnlyDebtor, receipt, bizEvent, debtorCF, payerCF, logger);
 
+
+                log = String.format(
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:139]: missing "eventId" biz

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 8]
```diff
@@ -161,22 +170,23 @@ public void processGenerateReceipt(
             itemsToNotify.add(receipt);
 
         } else {
-            discarder++;
+            String errorMessage = String.format(
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:173]: missing "eventId" biz

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java [hunk 9]
```diff
@@ -161,22 +170,23 @@ public void processGenerateReceipt(
...
-        logMsg = String.format("itemsDone stat %s function - number of PDFs sent to the receipt blob storage %d", context.getInvocationId(), numberOfSavedPdfs);
-        logger.info(logMsg);
 
-        //Call to datastore info
-        logMsg = String.format("GenerateReceiptProcess stat %s function - number of receipt inserted on the datastore %d", context.getInvocationId(), itemsToNotify.size());
-        logger.info(logMsg);
 
         if (!itemsToNotify.isEmpty()) {
+            String log = String.format(
+                    "[GenerateReceiptProcess] Receipt with id %s being saved with status %s and with %s pdfs",
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:184]: missing "eventId" biz

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java [hunk 10]
```diff
@@ -91,19 +91,20 @@ public void processManageReceiptPoisonQueue(
...
             }
         } else {
-            saveToDocument(errorMessage, documentdb);
+            saveToDocument(errorMessage, documentdb, logger);
         }
 
     }
 
-    private void saveToDocument(String errorMessage, OutputBinding<ReceiptError> documentdb) {
+    private void saveToDocument(String errorMessage, OutputBinding<ReceiptError> documentdb, Logger logger) {
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java:106]: missing "eventId" biz

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/service/BizEventToReceiptService.java [hunk 11]
```diff
@@ -57,7 +56,9 @@ public void handleSendMessageToQueue(BizEvent bizEvent, Receipt receipt, Logger
      */
     public void handleError(Receipt receipt) {
         receipt.setStatus(ReceiptStatusType.NOT_QUEUE_SENT);
-        ReasonError reasonError = new ReasonError(ReasonErrorCode.ERROR_QUEUE.getCode(), "Error sending message to queue");
+        ReasonError reasonError = new ReasonError(ReasonErrorCode.ERROR_QUEUE.getCode(),
+                String.format("[BizEventToReceiptService] Error sending message to queue" +
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/receipt/pdf/datastore/service/BizEventToReceiptService.java:60]: missing "eventId" bix

**Review by pasqualespica — CHANGES_REQUESTED** [review]
all logs shall : 
- start with fn name retrive from context
- contains/reference biz event id

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java [hunk 12]
```diff
@@ -52,7 +52,7 @@ public void processManageReceiptPoisonQueue(
         Logger logger = context.getLogger();
         BizEvent bizEvent = null;
 
-        String logMsg = String.format("ManageReceiptPoisonQueueProcessor function called at %s for payload %s",
+        String logMsg = String.format("[ManageReceiptPoisonQueueProcessor] function called at %s for payload %s",
```
[the remark this answers is not in the document]
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java:55]: At this stage the payload could be invalid

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/service/BizEventToReceiptService.java (hunk 11 above)
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/service/BizEventToReceiptService.java:60]: content  recovered from receipt.getEventId(), line 61

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java (hunk 10 above)
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/ManageReceiptPoisonQueue.java:106]: Could be generic invalid payload

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java (hunk 8 above)
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:177]: Done

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java (hunk 7 above)
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:141]: Done

File: src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java (hunk 6 above)
Comment (alessio-cialini) [reply src/main/java/it/gov/pagopa/receipt/pdf/datastore/GenerateReceiptPdf.java:130]: Done

**Review by pasqualespica — APPROVED** [review]

**Pull request merged**